agent/consul/leader_connect.go

```diff
@@ -146,6 +148,47 @@ func (s *Server) pruneCARoots() error {
 	return nil
 }
 
+func emitCAExpirationMetrics(s *Server) func(ctx context.Context) error {
+	key := []string{"mesh", "root-ca", "expiry"}
```

Comment:
Thoughts on this metric name? I couldn't find any other CA specific metrics. We could use a different key.